src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,334 @@
+ * breaking change.
+ * @param current current dependency version in project
+ * @param latest most up-to-date dependency version available on NPM
+ */
+function hasBreakingChange(current, latest) {
+  const currMajor = current.split('.')[0];
+  const latestMajor = latest.split('.')[0];
+  return +latestMajor - +currMajor >= 1;
+}
+
+/**
+ * Creates the update response - returns an object in the expected response format.
+ * @param versionMap version map of checked dependencies, current versions, and info on upgrade + breaking changes
+ * @param inaccessibleFiles array of files that could not be read or accessed
+ */
+function createUpdateResp(versionMap, inaccessibleFiles) {
+  const name = 'the Slack SDK';
+  const releases = [];
+  const message = '';
+  const url = 'https://api.slack.com/future/changelog';
+  const fileErrorMsg = createFileErrorMsg(inaccessibleFiles);
+
+  let error = null;
+  let errorMsg = '';
+
+  // Output information for each dependency
+  for (const sdk of Object.values(versionMap)) {
+    // Dependency has an update OR the fetch of update failed
+    if (sdk) {
+      releases.push(sdk);
+
+      // Add the dependency that failed to be fetched to the top-level error message
+      if (sdk.error && sdk.error.message) {
+        errorMsg += errorMsg
+          ? `, ${sdk}`
+          : `An error occurred fetching updates for the following packages: ${sdk.name}`;
+      }
+    }
+  }
+
+  // If there were issues accessing dependency files, append error message(s)
+  if (inaccessibleFiles.length) {
+    errorMsg += errorMsg ? `\n\n   ${fileErrorMsg}` : fileErrorMsg;
+  }
+
+  if (errorMsg) error = { message: errorMsg };
+
+  return {
+    name,
+    message,
+    releases,
+    url,
+    error,
+  };
+}
+
+/**
+ * Returns error when dependency files cannot be read.
+ * @param inaccessibleFiles array of files that could not be read or accessed
+ */
+function createFileErrorMsg(inaccessibleFiles) {
+  let fileErrorMsg = '';
+
+  // There were issues with reading some of the files that were found
+  for (const file of inaccessibleFiles) {
+    fileErrorMsg += fileErrorMsg
+      ? `\n   ${file.name}: ${file.error.message}`
+      : `An error occurred while reading the following files: \n\n   ${file.name}: ${file.error.message}`;
+  }
+
+  return fileErrorMsg;
+}
+/**
+ * Queries for current Deno version of the project.
+ */
+dependencyExports.getDenoCurrentVersion = async () => {
+  const { stdout } = await exec(
+    'npm list @slack/deno-slack-sdk --depth=0 --json'
+  ).catch((err) => {
+    throw new Error(err);
+  });
+
+  return stdout ? stdout : '';
+};
+
+/**
+ * Queries for current Bolt version of the project.
+ */
+dependencyExports.getBoltCurrentVersion = async () => {
+  const { stdout } = await exec(
+    'npm list @slack/bolt --depth=0 --json'
+  ).catch((err) => {
+    throw new Error(err, {
+      cause: err
+    });
+  });;
+
+  return stdout ? stdout : '';
```

Comment:
this is a great question! I originally was thinking if there is an error thrown then I want to prevent it from being returned as `stdout`, but you make a great point about how this feels unnecessary if it is empty. I'll remove this, great callout! 🙌